src/pages/Login.tsx

```diff
@@ -27,138 +25,60 @@ const Login: React.FC = () => {
     value: '',
   })
 
-  const [submissionStatus, setSubmissionStatus] = useState<SubmissionStatus>(
-    SubmissionStatus.yetToSubmit,
-  )
-
-  // TODO: Rework this using Redux Saga
-  async function submit() {
-    abortController?.abort()
-    abortController = new AbortController()
-
-    setSubmissionStatus(SubmissionStatus.submitting)
-
-    let userProfile: UserProfile
-    try {
-      await userService.createSession(
-        usernameFieldInfo.value,
-        passwordFieldInfo.value,
-        abortController,
-      )
-
-      userProfile = await userService.getUserProfile(abortController)
-    } catch (e) {
-      if (e instanceof CanceledError) {
-        setSubmissionStatus(SubmissionStatus.yetToSubmit)
-      } else if (e instanceof AxiosError && e.response?.status === 401) {
-        setSubmissionStatus(SubmissionStatus.failedErrorKnown)
-      } else {
-        setSubmissionStatus(SubmissionStatus.failedErrorUnknown)
-      }
-
-      return
-    }
-
-    setSubmissionStatus(SubmissionStatus.succeeded)
-    dispatch(updateUserInfo(userProfile))
-    navigate(Paths.Dashboard)
-  }
-
-  function getErrorMessage(
-    submissionStatus: SubmissionStatus,
-  ): string | undefined {
-    switch (submissionStatus) {
-      case SubmissionStatus.failedErrorKnown: {
-        return 'Incorrect username or password.'
-      }
-      case SubmissionStatus.failedErrorUnknown: {
-        return 'Sorry, please try again later.'
-      }
-      default: {
-        return undefined
-      }
-    }
-  }
+  const canSubmit: boolean =
+    !isSubmitting &&
+    usernameFieldInfo.value.length > 0 &&
+    passwordFieldInfo.value.length > 0 &&
+    usernameFieldInfo.errorMessage === undefined &&
+    passwordFieldInfo.errorMessage === undefined
 
   useEffect(() => {
-    return () => {
-      abortController?.abort()
+    if (isLoggedIn) {
+      navigate(Paths.Root)
     }
-  }, [])
+  }, [isLoggedIn]) // eslint-disable-line react-hooks/exhaustive-deps
+
+  async function handleSubmit(e: FormEvent) {
```

Comment:
Not sure if this function needs `async` since theres no `await`

Reply:
Yup, this was a mistake. Thanks for point it out.